libs/server/Resp/Objects/HashCommands.cs

```diff
@@ -67,6 +67,15 @@ private unsafe bool HashSet<TGarnetApi>(int count, byte* ptr, HashOperation hop,
 
                 var inputCount = (count - 1) / 2;
 
+                HashOperation hop = 0;
+                switch (command)
```

Comment:
Seems like many of these changes do not have to do with the ACL implementation. Unless I am mistaken but pushing the Hash operation type inside the method it will make it less clean and harder to do the ACL check globally. Similar to above the operation type should pass as parameter and if we want to do arity checks or ACL checks we should do it before reaching this point.

Reply:
This (and similar checks for List, SortedSet, and Geo operations) came out of removing the subcommand `byte` (not to be confused with Redis's notion of a subcommand) and hoisting those operations up into `RespCommand` to centralized ACL checking.

There are still some places where the `(List|Hash|Set|whatever)Operation` is needed, so while the "true" value so far as parsing is concerned is the `RespCommand` some new `switch`es (and changed `if`s) were added.